### ISSUE TITLE
Missing Illumination keys

### ISSUE THREAD

**Issue by juraj-visnovsky** [ceriath/candelafvtt#8]
Hello 👋, currently, we're missing Illumination keys related to the chosen Specialty.

I'd be happy to prepare a PR that adds these keys, but before I do that, I'd like to start a discussion on how this feature should work.

I believe it would make sense to include the Illumination Keys in the Specialties pack. However, I'm unsure about the level of flexibility this feature should offer for customization. Considering the openness of other features in this system, I think it makes sense to allow editing of the keys both in the Compendium pack and on the character sheet. However, I'm not entirely certain if it should be possible to change the number of Illumination keys by adding or removing them (as the Core Rulebook doesn't address this).

In summary, I would appreciate hearing your opinion on the following questions:

1. Should it be possible to edit Illumination keys belonging to a given Specialty in the Compendium pack?
2. Should it be possible to add/remove Illumination keys belonging to a given Specialty in the Compendium pack?
3. Should it be possible to edit Illumination keys on the character sheet?
4. Should it be possible to add/remove Illumination keys on the character sheet?

**Comment by ceriath** [comment]
Hi! 
Thank you for raising this as an issue. The illumnation keys have been on my mind too already, but i also did not have a clear idea on how and where to add them yet, so i left it open for later me (or someone else 😄). The questions stated in the way you did are really helping me though.
I'd say:
1. No
2. Yes
3. No
4. Yes

Elaboration: I think what we should do is add the illumination keys as an item type and in their own compendium. 
We would then link the default keys to the specialties, resulting in the keys being added to a character sheet, when the specialty is added. 
On the character sheet itself, they could then be removed again, in favor of another key (Q4). My reasoning behind that thought is, that i could imagine a player character having a special illumination key, that is tailored to that character (e.g. because of a personality trait, etc.). So removing one of the keys in favor of personal one (or adding a fourth personal key, we should not limit that and leave it up to the gamemaster), should be possible. 
I can not see any reason to edit the default illumination keys (Q1/3). I can't really think of a reason, why someone would want to do that. Instead, you would add your own custom illumination key and use that instead.
However, i think we need to enable gamemasters to add/remove keys to a specialty in the compendium (Q2), to support custom specialties. Otherwise there would be no way to link a key to a custom specialty.

We could probably just copy the "gear" item type and sheet and use it for illumination keys. We could then just copy the "gear" tab on the character sheet and should have most functionality. The only new thing that needs to be added is the default assignment from specialties and linking keys to a specialty. 

Any thoughts on that?

**Comment by juraj-visnovsky** [comment]
Hey @ceriath! 👋 Sorry for my late response.
Thank you for the clarification. I think the solution you outlined makes perfect sense. I'll prepare a PR soon. 🙂 

> The only new thing that needs to be added is the default assignment from specialties and linking keys to a specialty.

I have some doubts about this part. While I totally agree that it would be a nice feature to have, I'd only address it after we have the illumination keys in the system. Here's my reasoning:
- If we only  implement it for Illumination keys it would be inconsistent with how Gear works. Ideally, Gear, Illumination keys, and possibly Abilities (if we make them selectable) should be added automatically when a Specialty is selected.
- It could add additional complexity, as we would need to figure out what should happen with existing Illumination keys when assigning new Illumination keys from a Specialty. An easy solution might be to clear the old Illumination keys, but this could probably result in a bad UX (as it might delete user's custom illumination keys). What are your thoughts on this?

**Comment by juraj-visnovsky** [comment]
> We could then just copy the "gear" tab on the character sheet and should have most functionality.

By the way, what do you think about placing "Illumination keys" under "Background"? This tab won't be frequently used during a session, so I don't think it needs to occupy a prominent spot on the character sheet.

**Comment by ceriath** [comment]
Hi!
Sorry i didn't get back to you on this, i typed a reply on it, but apparently i never submitted it. :(
However, i did agree with your reasoning and i also do agree with having it in the Biography tab.
I'll have a look at the PR asap!
